var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -175,6 +177,14 @@ class Amdfftw(FftwBase):
         '+amd-app-opt',
         when='precision=quad',
         msg='Quad precision is not supported with amd-app-opt')
+    conflicts(
+        '+amd-dynamic-dispatcher',
+        when='%aocc',
+        msg='dynamic-dispatcher is not supported by AOCC clang compiler')
+    conflicts(
+        '+amd-dynamic-dispatcher',
+        when='@:3.1',
+        msg='dynamic-dispatcher is supported from 3.2 version onwards')
```

Comment:
This conflict can be removed making the variant conditional.

Reply:
Changes incorporated